lib/bundler/definition.rb

```diff
@@ -709,12 +709,8 @@ def converge_locked_specs
       # and Gemfile.lock. If the Gemfile modified a dependency, but
       # the gem in the Gemfile.lock still satisfies it, this is fine
       # too.
-      locked_deps_hash = @locked_deps.inject({}) do |hsh, dep|
-        hsh[dep] = dep
-        hsh
-      end
       @dependencies.each do |dep|
-        locked_dep = locked_deps_hash[dep]
```

Comment:
this might still need to be keyed off the dependency instead of just the name, it might be responsible for the test failures